packages/gdk_base.rb

```diff
@@ -18,19 +18,27 @@ def self.install
 
       # Do not edit this file. It will be overwritten by updates.
 
-      SCALE=1
-      [ -z "$DISPLAY" ] && DISPLAY=:0
-      [ -z "$GDK_BACKEND" ] && GDK_BACKEND=x11
-      RESOLUTION=$(xdpyinfo | awk '/dimensions:/ { print $2 }' | cut -d'x' -f1)
-      [[ $RESOLUTION -gt 1500 && $RESOLUTION -lt 2500 ]] && SCALE=1.5
-      [[ $RESOLUTION -ge 2500 && $RESOLUTION -lt 3500 ]] && SCALE=2
-      [[ $RESOLUTION -ge 3500 && $RESOLUTION -lt 4500 ]] && SCALE=2.5
-      [[ $RESOLUTION -ge 4500 && $RESOLUTION -lt 5500 ]] && SCALE=3
-      [[ $RESOLUTION -gt 5500 ]] && SCALE=3.5
-      export GDK_BACKEND=$GDK_BACKEND
-      export GDK_SCALE=$SCALE
-      export QT_SCALE_FACTOR=$SCALE
-      export DISPLAY=$DISPLAY
+      GDK_BACKEND=${GDK_BACKEND:-x11}
+      function roundhalves {
+                    echo "$1 * 2" | bc | xargs -I@ printf "%1.f" @ | xargs -I% echo "% * .5" | bc
+            }
+      pxwidth=$(WAYLAND_DISPLAY=wayland-0 wayland-info -i wl_output | grep width: | grep px | head -n 1 | awk '{print $2}')
+      lwidth=$(WAYLAND_DISPLAY=wayland-0 wayland-info -i zxdg_output_manager_v1 | grep logical_width:  | sed 's/,//' | awk '{print $2}')
+      # echo "pxwidth: $pxwidth, lwidth: $lwidth"
+      # SCALE needs to be rounded to the nearest 0.5
+      # Check to see if pxwidth and lwidth are integers before calculating SCALE.
+      # wayland-info on armv7l does not show lwidth, but aarch64 does.
+      if [[ $pxwidth == ?(-)+([0-9]) ]] && [[ $lwidth == ?(-)+([0-9]) ]] && [[ -z "$SCALE" ]] ; then
+        SCALE=$(roundhalves $(echo "scale=2 ;$lwidth / $pxwidth" | bc))
+      fi
+      #[[ $RESOLUTION -gt 1500 && $RESOLUTION -lt 2500 ]] && GDK_SCALE=1.5
+      #[[ $RESOLUTION -ge 2500 && $RESOLUTION -lt 3500 ]] && GDK_SCALE=2
+      #[[ $RESOLUTION -ge 3500 && $RESOLUTION -lt 4500 ]] && GDK_SCALE=2.5
+      #[[ $RESOLUTION -ge 4500 && $RESOLUTION -lt 5500 ]] && GDK_SCALE=3
+      #[[ $RESOLUTION -gt 5500 ]] && GDK_SCALE=3.5
+      SCALE=${SCALE:-1}
+      GDK_SCALE=$SCALE
+      QT_SCALE_FACTOR=$GDK_SCALE
```

Comment:
Why no `export`s?

Reply:
Because this line is in `/usr/local/etc/profile`:
```
# Export variables set (set allexport)
set -a
```
before
```
# Load the environment
for i in "${CREW_SYSCONFDIR}"/env.d/*; do [ -f "$i" ] && . "${i}"; done
```
So any variable set automatically gets exported.

Reply:
Gotcha.